# Marketing Campaign Analysis

Dataset info: <br/><br/>
This is a marketing campaign dataset from online source. <br/>
The main purpose of this project is to optimising the marketing strategy facing different clients classified by size, and to promote sales based on insights from it.


### 1. Data Understanding

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import scipy.stats
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
data=pd.read_csv('hospital_campaign_data.csv')
data.head()

Client ID      Client Type  Number of Customers  Montly Target  Zip Code  \
0  ID-987275  Medium Facility                 2800            125      1003   
1  ID-987275  Medium Facility                 2800            125      1003   
2  ID-987275  Medium Facility                 2800            125      1003   
3  ID-987275  Medium Facility                 2800            125      1003   
4  ID-987275  Medium Facility                 2800            125      1003   

  Calendardate  Amount Collected  Unit Sold  Campaign (Email)  \
0   16-01-2014                 0          0               0.0   
1   16-02-2014           3409460         24               0.0   
2   18-03-2014          10228384         75               0.0   
3   18-04-2014          17047304        123               0.0   
4   19-05-2014          23866224        171               0.0   

   Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  Sales Contact 2  \
0               0.0               0.0              0.0              0.0   
1               0.0               0.0              0.0              0.0   
2               0.0               0.0              0.0              0.0   
3               0.0               0.0              0.0        3547500.0   
4               0.0               0.0              0.0              0.0   

   Sales Contact 3  Sales Contact 4  Sales Contact 5 Number of Competition  
0              0.0              0.0              0.0                   Low  
1              0.0              0.0         322500.0                   Low  
2              0.0              0.0              0.0                   Low  
3        1290000.0              0.0              0.0                   Low  
4              0.0              0.0              0.0                   Low

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2976 entries, 0 to 2975
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Client ID              2976 non-null   object 
 1   Client Type            2976 non-null   object 
 2   Number of Customers    2976 non-null   int64  
 3   Montly Target          2976 non-null   int64  
 4   Zip Code               2976 non-null   int64  
 5   Calendardate           2976 non-null   object 
 6   Amount Collected       2976 non-null   int64  
 7   Unit Sold              2976 non-null   int64  
 8   Campaign (Email)       2976 non-null   float64
 9   Campaign (Flyer)       2976 non-null   float64
 10  Campaign (Phone)       2976 non-null   float64
 11  Sales Contact 1        2976 non-null   float64
 12  Sales Contact 2        2976 non-null   float64
 13  Sales Contact 3        2976 non-null   float64
 14  Sales Contact 4        2976 non-null   float64
 15  Sale

In [4]:
data.describe()

Number of Customers  Montly Target  Zip Code  Amount Collected  \
count          2976.000000    2976.000000    2976.0      2.976000e+03   
mean           1456.935484      75.080645    1003.0      1.700440e+07   
std            1669.848734      87.042114       0.0      3.025803e+07   
min               0.000000       5.000000    1003.0     -2.216150e+07   
25%             240.000000      10.000000    1003.0      0.000000e+00   
50%             960.000000      47.500000    1003.0      3.409460e+06   
75%            2090.000000     101.250000    1003.0      2.045676e+07   
max            9840.000000     510.000000    1003.0      2.079771e+08   

         Unit Sold  Campaign (Email)  Campaign (Flyer)  Campaign (Phone)  \
count  2976.000000      2.976000e+03      2.976000e+03      2.976000e+03   
mean    121.464718      1.432850e+05      6.854186e+05      2.977743e+04   
std     216.413984      7.230452e+05      1.727587e+06      3.832134e+05   
min     -63.000000      0.000000e+00      0.000000e+00      0.000000e+00   
25%       0.000000      0.000000e+00      0.000000e+00      0.000000e+00   
50%      24.000000      0.000000e+00      0.000000e+00      0.000000e+00   
75%     147.000000      0.000000e+00      8.148285e+04      0.000000e+00   
max    1500.000000      1.144673e+07      1.359395e+07      9.617380e+06   

       Sales Contact 1  Sales Contact 2  Sales Contact 3  Sales Contact 4  \
count     2.976000e+03     2.976000e+03     2.976000e+03     2.976000e+03   
mean      1.313407e+05     2.023319e+06     1.867487e+06     7.076361e+04   
std       1.034882e+06     3.396991e+06     3.271349e+06     3.869872e+05   
min       0.000000e+00     0.000000e+00     0.000000e+00     0.000000e+00   
25%       0.000000e+00     0.000000e+00     0.000000e+00     0.000000e+00   
50%       0.000000e+00     6.450000e+05     3.225000e+05     0.000000e+00   
75%       0.000000e+00     2.580000e+06     2.580000e+06     0.000000e+00   
max       2.902500e+07     3.063750e+07     2.612250e+07     4.837500e+06   

       Sales Contact 5  
count     2.976000e+03  
mean      1.462954e+04  
std       8.905955e+04  
min       0.000000e+00  
25%       0.000000e+00  
50%       0.000000e+00  
75%       0.000000e+00  
max       1.612500e+06

In [5]:
data['Client Type'].value_counts()

Large Facility      1368
Small Facility       840
Medium Facility      504
Private Facility     264
Name: Client Type, dtype: int64

In [6]:
pd.crosstab(data['Number of Competition'],data['Client Type'],margins=True,normalize='columns')

Client Type            Large Facility   Medium Facility  Private Facility  \
Number of Competition                                                       
High                          0.166667         0.166667          0.166667   
Low                           0.833333         0.833333          0.833333   

Client Type            Small Facility        All  
Number of Competition                             
High                          0.166667  0.166667  
Low                           0.833333  0.833333

In [7]:
data.groupby('Number of Competition').mean(numeric_only=True)

Number of Customers  Montly Target  Zip Code  \
Number of Competition                                                 
High                           1456.935484      75.080645    1003.0   
Low                            1456.935484      75.080645    1003.0   

                       Amount Collected   Unit Sold  Campaign (Email)  \
Number of Competition                                                   
High                       2.974789e+07  213.127016     105398.938508   
Low                        1.445570e+07  103.132258     150862.165766   

                       Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  \
Number of Competition                                                        
High                      994046.717540      45198.036895    146945.564516   
Low                       623692.979839      26693.304194    128219.758065   

                       Sales Contact 2  Sales Contact 3  Sales Contact 4  \
Number of Competition                                                      
High                      2.685333e+06     1.786754e+06     72172.379032   
Low                       1.890916e+06     1.883634e+06     70481.854839   

                       Sales Contact 5  
Number of Competition                   
High                       8452.620968  
Low                       15864.919355

In [8]:
data.groupby('Client Type').mean(numeric_only=True)

Number of Customers  Montly Target  Zip Code  \
Client Type                                                      
Large Facility            1380.842105      71.578947    1003.0   
Medium Facility           3940.761905     202.857143    1003.0   
Private Facility           400.727273      20.454545    1003.0   
Small Facility             422.514286      21.285714    1003.0   

                  Amount Collected   Unit Sold  Campaign (Email)  \
Client Type                                                        
Large Facility        1.999880e+07  143.098684     142273.609649   
Medium Facility       4.075997e+07  290.583333     437217.097817   
Private Facility      5.030246e+06   35.784091       5183.715152   
Small Facility        1.637759e+06   11.689286      11975.986310   

                  Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  \
Client Type                                                             
Large Facility        8.192056e+05      45595.436623    133667.763158   
Medium Facility       1.552603e+06      49176.847619    398645.833333   
Private Facility      2.272919e+05       5522.470455      1221.590909   
Small Facility        9.120875e+04          0.000000      8062.500000   

                  Sales Contact 2  Sales Contact 3  Sales Contact 4  \
Client Type                                                           
Large Facility       2.034013e+06     2.017039e+06    119287.280702   
Medium Facility      4.822783e+06     4.698646e+06     85104.166667   
Private Facility     6.376705e+05     4.434375e+05      3664.772727   
Small Facility       7.617143e+05     3.727946e+05      4223.214286   

                  Sales Contact 5  
Client Type                        
Large Facility       16266.447368  
Medium Facility      33273.809524  
Private Facility     12215.909091  
Small Facility        1535.714286

In [9]:
data['Calendardate']=pd.to_datetime(data['Calendardate'], dayfirst=True)
data['Calendar_Month']=data['Calendardate'].dt.month
data['Calendar_Year']=data['Calendardate'].dt.year

In [10]:
data.corr()['Amount Collected']

Number of Customers    0.607496
Montly Target          0.608204
Zip Code                    NaN
Amount Collected       1.000000
Unit Sold              0.997515
Campaign (Email)       0.248235
Campaign (Flyer)       0.444337
Campaign (Phone)       0.034858
Sales Contact 1        0.277478
Sales Contact 2        0.552112
Sales Contact 3        0.357887
Sales Contact 4        0.236165
Sales Contact 5        0.095795
Calendar_Month         0.139425
Calendar_Year          0.286194
Name: Amount Collected, dtype: float64

### 2.EDA

In [11]:
cm = sns.light_palette("orange", as_cmap=True)
correlation_analysis=pd.DataFrame(data[['Amount Collected',
'Campaign (Email)', 'Campaign (Flyer)', 'Campaign (Phone)',
       'Sales Contact 1', 'Sales Contact 2', 'Sales Contact 3',
       'Sales Contact 4', 'Sales Contact 5']].corr()['Amount Collected']).reset_index()
correlation_analysis.columns=['Impacting Variable','Degree of Linear Impact (Correlation)']
correlation_analysis=correlation_analysis[correlation_analysis['Impacting Variable']!='Amount Collected']
correlation_analysis=correlation_analysis.sort_values('Degree of Linear Impact (Correlation)',ascending=False)
correlation_analysis.style.background_gradient(cmap=cm).set_precision(2)

In [12]:
cm = sns.light_palette("orange", as_cmap=True)
correlation_analysis=pd.DataFrame(data.groupby('Client Type')[['Amount Collected',
       'Campaign (Email)', 'Campaign (Flyer)', 'Campaign (Phone)',
       'Sales Contact 1', 'Sales Contact 2', 'Sales Contact 3',
       'Sales Contact 4', 'Sales Contact 5']].corr()['Amount Collected']).reset_index()
correlation_analysis=correlation_analysis.sort_values(['Client Type','Amount Collected'],ascending=False)
correlation_analysis.columns=['Acc Type','Variable Impact on Sales','Impact']
correlation_analysis=correlation_analysis[correlation_analysis['Variable Impact on Sales']!='Amount Collected'].reset_index(drop=True)
correlation_analysis.style.background_gradient(cmap=cm).set_precision(2)

#### Test of Significance

In [13]:
import statsmodels.api as sm
import statsmodels.formula.api as smf
data.columns=[mystring.replace(" ", "_") for mystring in data.columns]
data.columns=[mystring.replace("(", "") for mystring in data.columns]
data.columns=[mystring.replace(")", "") for mystring in data.columns]
results = smf.ols('Amount_Collected ~ Campaign_Email+Campaign_Flyer+Campaign_Phone+\
       Sales_Contact_1 + Sales_Contact_2 + Sales_Contact_3+Sales_Contact_4 + Sales_Contact_5',data=data).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:       Amount_Collected   R-squared:                       0.480
Model:                            OLS   Adj. R-squared:                  0.478
Method:                 Least Squares   F-statistic:                     342.1
Date:                Tue, 19 Sep 2023   Prob (F-statistic):               0.00
Time:                        15:46:26   Log-Likelihood:                -54512.
No. Observations:                2976   AIC:                         1.090e+05
Df Residuals:                    2967   BIC:                         1.091e+05
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept        1.481e+06   5.12e+05     

In [14]:
df = pd.read_html(results.summary().tables[1].as_html(),header=0,index_col=0)[0]

In [15]:
df=df.reset_index()
df=df[df['P>|t|']<0.05][['index','coef']]
df

index          coef
0        Intercept  1.481000e+06
2   Campaign_Flyer  3.337600e+00
4  Sales_Contact_1  4.236800e+00
5  Sales_Contact_2  3.638200e+00
6  Sales_Contact_3  2.343200e+00
7  Sales_Contact_4  1.094780e+01

In [16]:
consolidated_summary=pd.DataFrame()
for acctype in list(set(list(data['Client_Type']))):
    temp_data=data[data['Client_Type']==acctype].copy()
    results = smf.ols('Amount_Collected ~ Campaign_Email+Campaign_Flyer+Campaign_Phone+\
       Sales_Contact_1 + Sales_Contact_2 + Sales_Contact_3+Sales_Contact_4 + Sales_Contact_5', data=temp_data).fit()
    df = pd.read_html(results.summary().tables[1].as_html(),header=0,index_col=0)[0].reset_index()
    df=df[df['P>|t|']<0.05][['index','coef']]
    df.columns=['Variable','Coefficent (Impact)']
    df['Account Type']=acctype
    df=df.sort_values('Coefficent (Impact)',ascending=False)
    df=df[df['Variable']!='Intercept']
    print(acctype)
    consolidated_summary=consolidated_summary.append(df)
    print(df)

    

Private Facility
          Variable  Coefficent (Impact)      Account Type
5  Sales_Contact_2               6.6223  Private Facility
Small Facility 
          Variable  Coefficent (Impact)     Account Type
5  Sales_Contact_2             0.810100  Small Facility 
3   Campaign_Phone             0.000008  Small Facility 
Large Facility 
          Variable  Coefficent (Impact)     Account Type
4  Sales_Contact_1              11.6731  Large Facility 
7  Sales_Contact_4              10.6145  Large Facility 
5  Sales_Contact_2               4.0031  Large Facility 
2   Campaign_Flyer               2.7204  Large Facility 
6  Sales_Contact_3               2.0316  Large Facility 
3   Campaign_Phone              -3.5361  Large Facility 
Medium Facility
          Variable  Coefficent (Impact)     Account Type
2   Campaign_Flyer               4.1059  Medium Facility
5  Sales_Contact_2               3.5778  Medium Facility
4  Sales_Contact_1               3.1365  Medium Facility
6  Sales_Contact_3   

In [17]:
import statsmodels.api as sm
import statsmodels.formula.api as smf
consolidated_summary=pd.DataFrame()
for acctype in list(set(list(data['Client_Type']))):
    print(acctype)
    temp_data=data[data['Client_Type']==acctype].copy()
    results = smf.ols('Amount_Collected ~ Campaign_Email+Campaign_Flyer+Campaign_Phone+\
       Sales_Contact_1 + Sales_Contact_2 + Sales_Contact_3+Sales_Contact_4 + Sales_Contact_5', data=temp_data).fit()
    df = pd.read_html(results.summary().tables[1].as_html(),header=0,index_col=0)[0].reset_index()
    df=df[df['P>|t|']<0.05][['index','coef']]
    df.columns=['Variable','Coefficent (Impact)']
    df['Account Type']=acctype
    df=df.sort_values('Coefficent (Impact)',ascending=False)
    df=df[df['Variable']!='Intercept']
    consolidated_summary=consolidated_summary.append(df)
    print(results.summary())

Private Facility
                            OLS Regression Results                            
Dep. Variable:       Amount_Collected   R-squared:                       0.346
Model:                            OLS   Adj. R-squared:                  0.326
Method:                 Least Squares   F-statistic:                     16.88
Date:                Tue, 19 Sep 2023   Prob (F-statistic):           4.63e-20
Time:                        15:46:30   Log-Likelihood:                -4650.8
No. Observations:                 264   AIC:                             9320.
Df Residuals:                     255   BIC:                             9352.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept        4.439e+0

### Insights and Recommendations

In [18]:
consolidated_summary

Variable  Coefficent (Impact)      Account Type
5  Sales_Contact_2             6.622300  Private Facility
5  Sales_Contact_2             0.810100   Small Facility 
3   Campaign_Phone             0.000008   Small Facility 
4  Sales_Contact_1            11.673100   Large Facility 
7  Sales_Contact_4            10.614500   Large Facility 
5  Sales_Contact_2             4.003100   Large Facility 
2   Campaign_Flyer             2.720400   Large Facility 
6  Sales_Contact_3             2.031600   Large Facility 
3   Campaign_Phone            -3.536100   Large Facility 
2   Campaign_Flyer             4.105900   Medium Facility
5  Sales_Contact_2             3.577800   Medium Facility
4  Sales_Contact_1             3.136500   Medium Facility
6  Sales_Contact_3             2.117400   Medium Facility

#### Insights:
From the results above, we can clearly find that for different account type, there are significant differences in effectiveness of various marketing channels. <br/>
For Large Facility Clients, each dollar's cost will value the most by Sales_Contact_1 and Sales_Contact_4, which bring 11.67 and 10.61 dollars each. However, the phone strategy for these clients may lead to a negative marketing effect.<br/>
For Medium Facility Clients, Flyer is the most efficient marketing channel, followed by Sales Contact 1-3, and there's no significant differences in them.<br/>

#### Recommendations:
Manage and reach out different type clients by different channel. <br/>
For Private and Small Facility Clients, invest mostly on Sales_Contact_2.<br/>
For For Large Facility Clients, use Sales_Contact_1 and Sales_Contact_4 and avoid phone channel for marketing.<br/>
For Medium Facility Clients, roughly spend evenly on Flyer, Sales_Contact_1 and Sales_Contact_2.<br/>
Also, when there's limited budget, it's better to spend less for marketing facing to Small Facilities.<br/>
In the meantime, keep collecting more data to further informed analysis.<br/>